routing/router.go

```diff
@@ -1832,7 +1834,30 @@ func (r *ChannelRouter) sendPayment(payment *LightningPayment,
 					paySession.ReportEdgeFailure(
 						failedEdge,
 					)
+					return
+				}
+
+				// Update routing hints.
+				var direction lnwire.ChanUpdateFlag
+				if failedEdge.direction == 1 {
+					direction = lnwire.ChanUpdateDirection
+				}
```

Comment:
`failedEdge.direction` can be used directly.

Reply:
Not sure. failedEdge direction is unit8 while Flags are unit16.

Reply:
Ok, you may need to do a type cast, but the if statement is not necessary.

Reply:
Done